pegen/pegen.c

```diff
@@ -30,12 +41,13 @@ raise_syntax_error(Parser *p, const char *errmsg, ...)
     } else {
         Py_INCREF(Py_None);
         filename = Py_None;
-        loc = PyUnicode_DecodeUTF8(p->tok->buf, (p->tok->cur)-(p->tok->buf), "replace");
+        loc = PyUnicode_FromString(p->tok->buf);
```

Comment:
What evidence do we have that `tok->buf` is always UTF8? It makes sense for it to be, but I'm not exactly thrilled about how the code in tokenizer.c is written, it's impossible to follow... Unless you have a quick insight this is not actionable, no need to research it further.

Reply:
> What evidence do we have that `tok->buf` is always UTF8?

I based this implementation in the text of PEP 263:

> Python's tokenizer/compiler combo will need to be updated to work as follows:
>* read the file
>* decode it into Unicode assuming a fixed per-file encoding
>* convert it into a UTF-8 byte string
>* tokenize the UTF-8 content
>* compile it, creating Unicode objects from the given Unicode data and creating string objects from the Unicode literal data by first reencoding the UTF-8 data into 8-bit string data using the given file encoding

It seems that at the point the tokenizer acts, is always utf8